chore(ci): remove paths ignore list in CI task

overlookmotel opened the pull request:
Remove the `paths-ignore` list from `CI` job.

It's ideal to skip running CI when a PR only touches `.md` files. But problem is that these jobs are marked as "required", so CI task *has* to run, or the PR can't be merged.

Maybe we can remove the requirement that these jobs complete before a PR can be merged, but for now this is the best solution.

overlookmotel commented:
* **#12749** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12749?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12749?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12749).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 reviewed (commented):
what about using https://github.com/oxc-project/oxc/blob/fe843806ece40a19adca3a7c6366abfb6f8662bb/.github/workflows/ci.yml#L153-L161 ?

from what i understand, the job still runs, it just exits early

overlookmotel commented:
That's a good idea, but I'm not completely sure how `dorny/paths-filter` works. Does adding `!xyz` to the list mean "return true if any file other than `xyz` has changed" or "return true if `xyz` hasn't changed"?

PRs which only alter `.md` files are quite rare, so I don't think it matters too much if CI runs on them when it doesn't need to. But what *does* matter is not making it really difficult to get those PRs merged.

So I'd propose merging this as is, and if we want to optimize it with `dorny/paths-filter` as you suggested, we can do that as a follow-on.

overlookmotel commented:
### Merge activity

* **Aug 2, 12:00 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
chore(ci): remove paths ignore list in CI task (#12749)

Remove the `paths-ignore` list from `CI` job.

It's ideal to skip running CI when a PR only touches `.md` files. But problem is that these jobs are marked as "required", so CI task *has* to run, or the PR can't be merged.

Maybe we can remove the requirement that these jobs complete before a PR can be merged, but for now this is the best solution.
